test/relayserver/RelayServer.test.ts

```diff
@@ -0,0 +1,1259 @@
+import { ethers } from 'hardhat';
+import { expect } from 'chai';
+import {
+  createEnvelopingTxRequest,
+  createEnvelopingRequest,
+  getServerInstance,
+  getInitiatedServer,
+} from './ServerTestEnvironments';
+import {
+  DeployVerifier,
+  RelayHub,
+  RelayVerifier,
+} from '@rsksmart/rif-relay-contracts';
+import {
+  buildServerUrl,
+  createSupportedSmartWallet,
+  deployRelayHub,
+  SupportedSmartWallet,
+  evmMineMany,
+  deploySmartWalletContracts,
+} from '../utils/TestUtils';
+import config from 'config';
+import {
+  AppConfig,
+  BlockchainConfig,
+  defaultEnvironment,
+  RelayServer,
+  SendTransactionDetails,
+  ServerAction,
+  ServerConfigParams,
+  SignedTransactionDetails,
+} from '@rsksmart/rif-relay-server';
+import {
+  AccountManager,
+  estimateInternalCallGas,
+  estimateTokenTransferGas,
+  HubInfo,
+  RelayClient,
+  RelayRequest,
+  RelayRequestBody,
+  setEnvelopingConfig,
+  UserDefinedRelayRequest,
+} from '@rsksmart/rif-relay-client';
+import { BigNumber, constants, Wallet } from 'ethers';
+import { spy, match } from 'sinon';
+import {
+  TestDeployVerifierConfigurableMisbehavior,
+  TestRecipient,
+  TestVerifierConfigurableMisbehavior,
+} from '../../typechain-types';
+import {
+  assertEventHub,
+  getTotalTxCosts,
+  loadConfiguration,
+  stringifyEnvelopingTx,
+} from './ServerTestUtils';
+import { mintTokens, prepareToken } from '../smartwallet/utils';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { sleep } from '@rsksmart/rif-relay-server';
+
+const SERVER_WORK_DIR = './tmp/enveloping/test/server';
+
+const basicAppConfig: Partial<AppConfig> = {
+  checkInterval: 10,
+  logLevel: 5,
+  workdir: SERVER_WORK_DIR,
+};
+
+const originalConfig = config.util.toObject(config) as ServerConfigParams;
+
+describe('RelayServer', function () {
+  describe('init', function () {
+    beforeEach(async function () {
+      const relayHub = await deployRelayHub();
+      loadConfiguration({
+        app: basicAppConfig,
+        contracts: {
+          relayHubAddress: relayHub.address,
+        },
+      });
+    });
+
+    afterEach(function () {
+      config.util.extendDeep(config, originalConfig);
+    });
+
+    it('should initialize relay params (chainId, networkId, gasPrice)', async function () {
+      const [relayOwner] = await ethers.getSigners();
+
+      const { chainId } = ethers.provider.network;
```

Comment:
Over here I could create a global variable holding the provider instead of accessing using ethers from hardhat.

Reply:
updated